### PR TITLE
skip step 1 of sign up if email is valid and terms have been agreed to

### PR DESCRIPTION
**Pull request by ScottDowne**

### CODE REVIEW

**Comment by ScottDowne** [comment]
#139

**Comment by ScottDowne** [comment]
jbuck r?

File: dist/ngWebmakerLogin.js [hunk 1]
```diff
@@ -1267,6 +1267,9 @@ ngModule.directive('wmJoinWebmaker', [
                 username: function () {
                   return username;
                 },
+                agreeToTerms: function () {
```
Comment (cadecairos) [inline dist/ngWebmakerLogin.js]: can you move this below the schowCTA function?

File: dist/ngWebmakerLogin.js [hunk 2]
```diff
@@ -1293,13 +1296,16 @@ ngModule.directive('wmJoinWebmaker', [
 
             if (email) {
               $scope.user.email = email;
-              joinController.validateEmail(email);
```
Comment (cadecairos) [inline dist/ngWebmakerLogin.js]: is there a reason you removed this line?

Comment (ScottDowne) [inline dist/ngWebmakerLogin.js]: I moved it to later in the process, it now validates inside "displayEmailInput" event.

https://github.com/mozilla/webmaker-login-ux/pull/143/files#diff-cedb0953ab67cb6defeeaf13421801a6R1317

Comment (ScottDowne) [inline dist/ngWebmakerLogin.js]: I could leave it where it was, and store the result then use it later, but didn't seem necessary.

**Comment by ScottDowne** [comment]
I'm using the test pages now, much easier for testing.

I added the same feature to plain. I used these for testing:

```
          client.create();
          //client.create("fwaf@fwaf.com", "", true);
          //client.create(undefined, "", true);
          //client.create("fsegesg", "", true);
          //client.create("fwaf@fwaf.com", "", false);
          //client.create("fwaf", "", false);
          //client.create("fwaf@fwaf.com", "fesgs", true);
          //client.create("fwaf", "fesgs", true);
          //client.create("fwaf@fwaf.com", "fesgs", false);
          //client.create("fwaf", "fesgs", false);
```

Each of these should fail or pass as follow:
1. should not show and errors and display an empty step 1.
2. should skip to step 2 and have no username filled in, no errors.
3. should not show and errors and display an empty step 1.
4. should show step 1 with an error on email.
5. should show step 1 with an error to accept terms and agreements.
6. should show step 1 with an error on email.
7. should show step 2 with a username filled in
8. should show step 1 with an error under email
9. should show step 1 with an error on terms and agreement
10. should show step 1 with an error on email

I'm going to do a similar set of tests for function calls for the angualr plugin.

**Comment by cadecairos** [comment]
The warning message for the email input appears when the join dialog opens. I confirmed that your patch introduced this behaviour) to see it, go to localhost:4321 and hit join webmaker

**Comment by ScottDowne** [comment]
Updates to handle that same list of tests cept with angular:

```

        $rootScope.joinWebmaker($('.home-email-field').val());
        //$rootScope.joinWebmaker();
        //$rootScope.joinWebmaker("fesgs@fawfa.com", "", true);
        //$rootScope.joinWebmaker(undefined, "", true);
        //$rootScope.joinWebmaker("fesfs", "", true);
        //$rootScope.joinWebmaker("fesfes@fwaf.com", "", false);
        //$rootScope.joinWebmaker("sefesg", "", false);
        //$rootScope.joinWebmaker("fesgs@fawfa.com", "sfesgs", true);
        //$rootScope.joinWebmaker("fesfs", "fesgs", true);
        //$rootScope.joinWebmaker("fesfes@fwaf.com", "esseg", false);
        //$rootScope.joinWebmaker("sefesg", "fesgs", false);
```

I did this from inside webmaker.org because I was familiar with it, but I am sure there is a similar way using the angular-test.html page

**Comment by ScottDowne** [comment]
I was able to fix #140 in this too.

It was blocked on this one, so I figured they are connected enough to be one bug.

**Comment by cadecairos** [comment]
r+

**Pull request merged**